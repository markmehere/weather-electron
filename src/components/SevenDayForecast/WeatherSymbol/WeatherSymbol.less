.weatherSymbol {
  width: 15vw;
}

.weatherSymbol img {
  width: 50%;
}

.weatherDescription {
  font-size: 85%;
  color: #888;
  margin: 0;

  &Small {
    font-size: 75%;
    color: #ccc;
    margin: 0;
  }
}
