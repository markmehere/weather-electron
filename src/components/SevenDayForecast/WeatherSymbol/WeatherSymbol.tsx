import React from "react";
import weatherAbbr from "../../../data/weather-abbr.json";
import styles from "./WeatherSymbol.less";
import { WeatherKind } from "../../../types/WeatherKind.js";
import { getStatic } from "../../../static";
import moment from "moment";

export interface WeatherSymbolProps {

  // The kind of weather
  kind: WeatherKind;

  // The date of this weather
  date: string;

}

export class WeatherSymbol extends React.Component<WeatherSymbolProps> {

  getSymbol = (kind: WeatherKind) => getStatic(`${kind as string}.svg`);

  getHumanReadable = (kind: WeatherKind) => weatherAbbr[kind];

  render() {
    const prettyDate = moment(this.props.date).format('dddd');
    return <div className={styles.weatherSymbol}>
      <img src={this.getSymbol(this.props.kind)}
        alt={this.getHumanReadable(this.props.kind)} />
      <p className={styles.weatherDescriptionSmall}>{prettyDate}</p>
      <p className={styles.weatherDescription}>{this.getHumanReadable(this.props.kind)}</p>
    </div>;
  }

}