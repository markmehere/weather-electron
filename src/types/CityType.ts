export type CityType = {
  value: number; /* woeid */
  label: string; /* city, country code (e.g. Melbourne, AU) */
};