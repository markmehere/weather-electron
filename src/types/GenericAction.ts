import { Action } from "redux";

export interface GenericAction<T> extends Action {
  payload: T;
} 
