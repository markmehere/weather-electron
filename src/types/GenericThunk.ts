import { ThunkAction } from "redux-thunk";
import { AppStore } from "../redux/reducers";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type GenericThunk = ThunkAction<any, AppStore, undefined, any>;
