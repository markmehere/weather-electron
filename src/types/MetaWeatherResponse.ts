import { WeatherKind } from "./WeatherKind";

export interface ConsolidatedWeatherResponse {
  id: string;
  weather_state_name: string;
  weather_state_abbr: WeatherKind;
  created: string;
  applicable_date: string;
  min_temp: number;
  max_temp: number;
  wind_speed: number;
  wind_direction: number;
  air_pressure: number;
  humidity: number;
  visibility: number;
  predictability: number;
}

export interface MetaWeatherResponse {
  consolidated_weather: ConsolidatedWeatherResponse[]
}