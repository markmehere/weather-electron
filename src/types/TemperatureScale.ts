export enum TemperatureScale {
  Celsius,
  Fahrenheit
}
