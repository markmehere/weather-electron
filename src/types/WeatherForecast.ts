import { WeatherKind } from "./WeatherKind";

export interface WeatherForecast {
  max: number;
  min: number;
  date: string;
  kind: WeatherKind;
}
