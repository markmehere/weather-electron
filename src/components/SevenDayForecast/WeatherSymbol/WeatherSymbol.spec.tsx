import React from 'react';
import { shallow } from 'enzyme';
import { WeatherKind } from '../../../types/WeatherKind';
import { WeatherSymbol } from './WeatherSymbol';

describe('<WeatherSymbol />', () => {

  const defaultProps = {
    kind: WeatherKind.Clear,
    date: '2020-05-10'
  };

  it('should formulate correct image link', () => {
    const mockWeatherSymbol = new WeatherSymbol(defaultProps);
    expect(mockWeatherSymbol.getSymbol('testme' as any)).toBe('http://localhost/static/testme.svg');
  });

  it('should formulate correct human readable string', () => {
    const mockWeatherSymbol = new WeatherSymbol(defaultProps);
    expect(mockWeatherSymbol.getHumanReadable(WeatherKind.Clear)).toBe('Clear');
    expect(mockWeatherSymbol.getHumanReadable(WeatherKind.LightCloud)).toBe('Light Cloud');
    expect(mockWeatherSymbol.getHumanReadable(WeatherKind.HeavyRain)).toBe('Heavy Rain');
    expect(mockWeatherSymbol.getHumanReadable(WeatherKind.Sleet)).toBe('Sleet');
    expect(mockWeatherSymbol.getHumanReadable(WeatherKind.Snow)).toBe('Snow');

  });

  it('matches snapshot', () => {
    const result = shallow(<WeatherSymbol {...defaultProps}  />);
    expect(result).toMatchSnapshot();
  });

});
