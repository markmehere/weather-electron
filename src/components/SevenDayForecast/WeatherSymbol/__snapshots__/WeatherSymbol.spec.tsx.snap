// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherSymbol /> matches snapshot 1`] = `
<div
  className="weatherSymbol"
>
  <img
    alt="Clear"
    src="http://localhost/static/c.svg"
  />
  <p
    className="weatherDescriptionSmall"
  >
    Sunday
  </p>
  <p
    className="weatherDescription"
  >
    Clear
  </p>
</div>
`;
